ReactiveCocoa/Swift/Signal.swift

```diff
@@ -151,54 +173,32 @@ public final class Signal<T, E: ErrorType> {
 	public func observe(error error: (E -> ())? = nil, completed: (() -> ())? = nil, interrupted: (() -> ())? = nil, next: (T -> ())? = nil) -> Disposable? {
 		return observe(Event.sink(next: next, error: error, completed: completed, interrupted: interrupted))
 	}
-}
-
-infix operator |> {
-	associativity left
-
-	// Bind tighter than assignment, but looser than everything else.
-	precedence 95
-}
-
-/// Applies a Signal operator to a Signal.
-///
-/// Example:
-///
-/// 	intSignal
-/// 	|> filter { num in num % 2 == 0 }
-/// 	|> map(toString)
-/// 	|> observe(next: { string in println(string) })
-public func |> <T, E, X>(signal: Signal<T, E>, @noescape transform: Signal<T, E> -> X) -> X {
-	return transform(signal)
-}
 
-/// Maps each value in the signal to a new value.
-public func map<T, U, E>(transform: T -> U) -> Signal<T, E> -> Signal<U, E> {
-	return { signal in
-		return Signal { observer in
-			return signal.observe { event in
+	/// Maps each value in the signal to a new value.
+	@warn_unused_result(message="Did you forget to call `observe` on the signal?")
+	public func map<U>(transform: T -> U) -> Signal<U, E> {
```

Comment:
Same here and for all these operators—can we extend `SignalType` instead of `Signal` directly?